rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1105,3 +1130,29 @@ private module Cached {
 }
 
 import Cached
+
+cached
+private module OptionalSteps {
```

Comment:
This should be inside the existing `Cached` module to make sure that they are evaluated together.